spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -54,18 +63,16 @@
  */
 public class DirectMessageListenerContainer extends AbstractMessageListenerContainer {
 
-	private final List<SimpleConsumer> consumers = new ArrayList<SimpleConsumer>();
+	private final List<SimpleConsumer> consumers = new LinkedList<SimpleConsumer>();
+
+	private final MultiValueMap<String, SimpleConsumer> consumersByQueue = new LinkedMultiValueMap<>();
 
 	private final MessagePropertiesConverter messagePropertiesConverter = new DefaultMessagePropertiesConverter();
 
 	private volatile TaskExecutor taskExecutor;
 
 	private volatile int consumersPerQueue = 1;
 
-	private volatile int prefetch = 1;
-
-	private volatile boolean defaultRequeueRejected = false;
-
 	private volatile long recoveryInterval;
```

Comment:
![recoveryinterval_usage](https://cloud.githubusercontent.com/assets/1071590/18331341/26840f12-752c-11e6-98c0-21a4f3deefba.png)

I wonder if we can move `recoveryBackOff` option from the `SimpleMessageListenerContainer` to the super `AbstractMessageListenerContainer`.